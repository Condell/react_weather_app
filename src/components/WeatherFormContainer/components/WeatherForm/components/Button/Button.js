import styled from "styled-components";
import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const Button = styled(Link)`
  font-family: "Arial";
  text-decoration: none;
  background-color: #4caf50;
  color: white;
  border-radius: 5px;
  padding: .5em;
  margin: .5em;
`;

Button.propTypes = {
  to: PropTypes.object
}

Button.defaultProps = {
  to: {}
}

export default Button;
