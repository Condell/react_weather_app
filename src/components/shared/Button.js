import React from 'react';
import styled from 'styled-components';

const Button = styled.button`
    background-color: #4CAF50;
    color: white;
    border-radius: 5px;
    padding: .5em 2em;
`;

export default Button;