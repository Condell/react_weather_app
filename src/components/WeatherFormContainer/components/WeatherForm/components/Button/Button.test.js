import React from "react";
import { shallow } from "enzyme";
import 'jest-styled-components'
import toJson from "enzyme-to-json"

import Button from "./Button";


describe("Button", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Button />);
  });

  it("should render", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
