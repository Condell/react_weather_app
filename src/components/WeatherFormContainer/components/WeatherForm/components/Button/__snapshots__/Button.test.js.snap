// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
.bZMiZE {
  font-family: "Arial";
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #4caf50;
  color: white;
  border-radius: 5px;
  padding: .5em;
  margin: .5em;
}

<Link
  className="sc-bdVaJa bZMiZE"
  replace={false}
  to={Object {}}
/>
`;
